docs/vagrant.md

```diff
@@ -0,0 +1,132 @@
+# Automated Ubuntu Local Env Setup
+
+You can optionally setup a virtual Ubuntu environment using Vagrant. It is a tool used for building and managing virtual machine environments in a single workflow.
+
+Although Vagrant supports [VirtualBox](https://www.virtualbox.org/), [Hyper-V](https://learn.microsoft.com/en-us/virtualization/hyper-v-on-windows/quick-start/enable-hyper-v), [Docker](https://www.docker.com/), or [VMware Fusion](https://customerconnect.vmware.com/downloads/get-download?downloadGroup=FUS-PUBTP-2021H1), `VirtualBox` is the only environment that has been developed and tested thus far. If you have need to use Hyper-V, Docker, or VMware, please create an issue and request a new feature.
+
+## Notes
+
+This was tested using:
+
+- Vagrant v2.4.0
+- VirutalBox v7.0.12
+
+## Prerequisites
+
+1. You will require 4 GB of memory.
+1. You will require 8 GB of disk space.
+1. Ports 4200, 4300, and 5432 must be available on host environment. If you have powershell installed you can use this command to list active ports:
+
+   ```powershell
+   Get-NetTCPConnection | Where-Object { $_.State -eq 'Listen' }
+   ```
+
+1. `npm install` creates symbolic links. As a result, the VM user: `vagrant` requires `create symbolic link` authority. In some environements, only the administrator has this authority. To resolve this issue in Windows, the following steps can be performed:
+   1. Launch the `Local Security Policy` app from the `start menu` (or `Win+R`, then type `secpol.msc`)
+   1. Navigate to `Local Policies` -> `User Rights Assignment`.
+   1. Open the `Create symbolic links` property. By default it has only `Administrators` listed.
+   1. Click `Add User or Group...` button and type `Authenticated Users` in the `Enter the object names to select` field and then press the `OK` button. This will grant `symbolic link` priviledge to all users who log on with credetials.
+   1. Close any open windows by clicking `OKs` and and then `log off` and `log on` to activate.
+   1. Symbolic links should now work in VirtualBox shared folders without administrator priviledges.
+
+## Setup Steps
+
+1. [Download](https://developer.hashicorp.com/vagrant/downloads?ajs_aid=ba208f95-d5a1-457d-abeb-49d458b95eec&product_intent=vagrant) Vagrant software.
+1. Reference these [installation](https://developer.hashicorp.com/vagrant/docs/installation) instructions.
+1. Verify installation by running `vagrant -v` in a terminal session.
+1. Install [VirtualBox](https://www.virtualbox.org/).
+1. If you have not already done so, fork repo to your local machine.
+1. Vagrant uses a special configuration file called `Vagrantfile` that contains all the information required for the creation of a Virtual Machine (guest) on your workstation (host) . It is located n your repo root directory.
```

Comment:
```suggestion
1. Vagrant uses a special configuration file called `Vagrantfile` that contains all the information required for the creation of a Virtual Machine (guest) on your workstation (host). It is located in your repo root directory.
```

You missed a few things in the last correction.

Reply:
You can use the "Commit suggestion" button to avoid missing anything.

Reply:
Never noticed that button.  Thanks

Reply:
This is a response to your comment below:

How long does it usually take to boot the VM? This is at the bottom of the output of `vagrant up`.

```text
==> default: Waiting for machine to boot. This may take a few minutes...
    default: SSH address: 127.0.0.1:2222
    default: SSH username: vagrant
    default: SSH auth method: private key
```

I have experienced this issue a number of times as I was working this PR. It appears to be a bug in the relationship between vagrant and virtualbox.  At this point in the process, the VM is up and running and vagrant is trying to establish a SSH session to it. However, it hangs indefinitely at this point and vagrant eventually times out. There is a vagrantfile line `config.vm.boot_timeout` were you can increase the timeout value, however, this would not solve the problem as you would just have to wait long before it times out.  

I don't know why this happens. And it does not happen every time. I notice that it happens consistently when I perform a `vagrant halt`, reboot the host, and then issue a `vagrant up`.

If you find that vagrant is stuck at this point, you can free it up by performing the following steps:
1. Open `Oracle VM VirtualBox Manager`
2. Click your VM machine in the left panel. The name starts with the directory name of the root of your repository. In our case it would be `gloss-translation_....`.
3. If the VM is running, you will see a green right arrow called `Show`. It opens a new panel where you can monitor the boot process of your VM.  For some unknown reason, when I do this, it frees up the vagrant session and then continue booting up to completion.
4. Once the vagrant session is up, you can close down the panel, without shutting down the VM by clicking menu `Machine/Detach GUI...`.

This was one of the reasons why I thought we should let developers know that this tool exists.  It does have a lot of useful features in it.

Reply:
Yeah go ahead and add these steps to the caveat section. They could be useful for future devs.

Reply:
I'm guessing this will also be true for anyone who has virtualbox installed for them by vagrant. This is probably because we don't need the full suite of virtualbox tools installed to use vagrant.

Reply:
I think we can leave this be for now. It seems like vagrant integration with virtualbox has some common issues. I look forward to using the docker configuration instead. I think it might be the better option since we already use docker in this project for other things.